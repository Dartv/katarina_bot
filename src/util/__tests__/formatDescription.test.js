const { formatDescription } = require('../');

describe('formatDescription', () => {
  const commandName = 'test';
  const description = 'description';
  const usage = '<param>';

  beforeEach(() => {
    process.env.BOT_PREFIX = '!';
  });

  it('should return raw string if argument is a string', () => {
    expect(formatDescription(description)).toBe(description);
  });

  it('should return just a description if no "usage" property provided', () => {
    expect(formatDescription({ description })).toBe(description);
  });

  it('should properly format description', () => {
    const actual = formatDescription({ commandName, description, usage });
    expect(actual).toMatchSnapshot();
  });
});
