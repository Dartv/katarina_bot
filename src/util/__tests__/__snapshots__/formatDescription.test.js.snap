// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatDescription should properly format description 1`] = `
"description

usage: \`!test <param>\`"
`;
